component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/util/EventMapAttributeFilter.java

```diff
@@ -0,0 +1,45 @@
+package org.wso2.am.analytics.publisher.util;
+
+import org.wso2.am.analytics.publisher.exception.MetricReportingException;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+public class EventMapAttributeFilter {
+    private static  final EventMapAttributeFilter INSTANCE = new EventMapAttributeFilter();
+
+    public static EventMapAttributeFilter getInstance(){
+        return INSTANCE;
+    }
+
+    public Map<String,Object> filter(Map<String,Object> source, Map<String,Class> requiredAttributes) throws  MetricReportingException{
+
+        Set<String> targetKeys = requiredAttributes.keySet();
```

Comment:
This functionality is not needed. During the validation we can check whether the required attributes are present. And at the build state its just going
filteredMap.put(targetKey, source.get(targetKey))

Reply:
Again this is regarding ELK  Tishan..
In case of ELK it doesn't check if required attributes are available in the validation phase.

Reply:
+ Tishan, can we rely on the fact that when one writes a new event builder.. let's think datadog.. the dev will include this validation explicitly in the validation stage, what if he end up implementing the validate() function just like in ELK scenario.

Reply:
The model we have provided for them is to use validation method to validate. Thats is precisely why build is a private method which will be only called after validation. We have enforced that. Hence build method should not have validations inside. and we should not change our interfaces to facilitate such

Reply:
Noted Tishan.
I will change it..
Modification i'm making here will resolve the exception throwing issue.